test/util/periods_unit.js

```diff
@@ -459,6 +459,19 @@ describe('PeriodCombiner', () => {
     a3.bandwidth = 97065;
     a2.roles = ['role1', 'role2'];
 
+    // t1 and t3 are duplicats
```

Comment:
Minor nit: typo on "duplicates"

Larger concern: the test has no expectations on the text streams, so it would pass even without your code change.  Please add a check to the end of the test to show that t3 was filtered out.  Thanks!

Reply:
Didn't see that one 😅